src/Container.ts

```diff
@@ -421,7 +421,7 @@ export class Container<Services = {}> {
   providesClass = <Token extends TokenType, Service, Tokens extends readonly ValidTokens<Services>[]>(
     token: Token,
     cls: InjectableClass<Services, Service, Tokens>
-  ) => this.providesService(ClassInjectable(token, cls));
+  ) => this.providesService(ClassInjectable(token, cls)) as Container<AddService<Services, Token, Service>>;
```

Comment:
I would prefer if we have return type explicitly declared on this method. There is a eslint rule for this, I'm thinking to enable it. That will not solve our problem, but that will prevent similar issues to happen in future when TS infers type incorrectly.

Reply:
Done! https://github.com/Snapchat/ts-inject/pull/14/commits/93365db0134fb2abbfceb660c3327e52df49d81f